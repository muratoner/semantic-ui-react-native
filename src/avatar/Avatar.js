import React, { useMemo, useState } from 'react'
import {
	ActivityIndicator,
	ImageBackground,
	StyleSheet,
	Text,
	TouchableOpacity,
	View
} from 'react-native'
import Icon from '../icons/Icon'
import { AvatarProps } from '../index'
import { Styles } from '../mixins'
import { UtilColor } from '../utils'

const Component = ({
	title,
	source,
	size = 32,
	titleShowLimit,
	style,
	containerStyle,
	icon,
	circular = true,
	rounded,
	withRandomColor = true,
	shortestTitle = true,
	onPress,
	textColorMode,
	bordered,
	raised,
	loading = true
}: AvatarProps) => {
	const [avatarLoaded, setAvatarLoaded] = useState(false)
	const [defaultBg] = useState('#b1b1b1')
	const [defaultTextColor] = useState('#fff')
	const chars = useMemo(() => {
		if (!shortestTitle) return title
		let chars = title
			?.trim()
			?.split(' ')
			.map((word) => word?.trim()?.charAt(0))
		if (titleShowLimit) {
			return chars.slice(0, titleShowLimit)
		}
		return chars
	}, [title, titleShowLimit, shortestTitle])

	const Wrapper = useMemo(() => {
		if (onPress) {
			return TouchableOpacity
		} else {
			return View
		}
	}, [onPress])

	const ComputedStyle = useMemo(() => {
		let style = {}
		bordered && (style = {...style, ...Styles.border})
		raised && (style = {...style, ...Styles.boxShadow})
		return style
	}, [bordered, raised])

	const avatarTitleAutoColor = useMemo(() => {
		if (textColorMode == 'light') {
			return UtilColor.lightenColor(UtilColor.stringToColour(title), 100)
		} else {
			return UtilColor.contrast(UtilColor.stringToColour(title))
		}
	}, [textColorMode, title])

	const avatarTitleContent = useMemo(
		() => (
			<Text
				minimumFontScale={0.01}
				adjustsFontSizeToFit
				numberOfLines={1}
				style={StyleSheet.flatten([
					styles.text(
						size,
						source || !withRandomColor ? defaultTextColor : avatarTitleAutoColor
					),
					style
				])}
			>
				{chars}
			</Text>
		),
		[
			title,
			size,
			withRandomColor,
			source,
			shortestTitle,
			titleShowLimit,
			avatarTitleAutoColor
		]
	)

	const avatarTitle = useMemo(
		() => (
			<Wrapper
				onPress={onPress}
				style={[
					styles.textContainer(
						size,
						withRandomColor ? UtilColor.stringToColour(title) : defaultBg,
						circular,
						rounded
					),
					ComputedStyle
				]}
			>
				{avatarTitleContent}
			</Wrapper>
		),
		[
			title,
			size,
			circular,
			withRandomColor,
			Wrapper,
			avatarTitleContent,
			ComputedStyle
		]
	)

	const avatarIcon = useMemo(() => {
		return (
			<Wrapper
				onPress={onPress}
				style={[
					styles.textContainer(size, defaultBg, circular, rounded),
					ComputedStyle
				]}
			>
				<Icon
					minimumFontScale={0.01}
					adjustsFontSizeToFit
					numberOfLines={1}
					style={StyleSheet.flatten([styles.icon(size, defaultTextColor), style])}
					{...icon}
				/>
			</Wrapper>
		)
	}, [size, icon, circular, Wrapper, ComputedStyle])

	const avatarImage = useMemo(
		() => (
			<Wrapper
				onPress={onPress}
				style={[
					ComputedStyle,
					styles.imageContainer(size, circular, rounded),
					containerStyle
				]}
			>
				<ImageBackground
					source={source}
					style={StyleSheet.flatten([styles.image(size, circular, rounded), style])}
					resizeMode='cover'
					onLoadEnd={() => {
						setAvatarLoaded(true)
					}}
				>
					{title && !avatarLoaded && avatarTitleContent}
					{!avatarLoaded && loading && (
						<ActivityIndicator color='#fff' style={styles.indicator} />
					)}
				</ImageBackground>
			</Wrapper>
		),
		[
			source,
			size,
			avatarLoaded,
			circular,
			title,
			Wrapper,
			avatarTitleContent,
			ComputedStyle,
			loading
		]
	)

	const avatar = useMemo(() => {
		if (source) {
			return avatarImage
		} else if (icon) {
			return avatarIcon
		} else {
			return avatarTitle
		}
	}, [avatarTitle, avatarIcon, avatarImage])

	return avatar
}

const styles = StyleSheet.create({
	indicator: {
		position: 'absolute',
		backgroundColor: 'rgba(0,0,0,.5)',
		width: '100%',
		height: '100%'
	},
	icon: (size, color) => ({
		color,
		textAlignVertical: 'center',
		textAlign: 'center',
		fontSize: size / 1.75
	}),
	text: (size, color) => ({
		color,
		textAlignVertical: 'center',
		textAlign: 'center',
		fontSize: 200,
		paddingHorizontal: size / 7
	}),
	imageContainer: (size, radius, rounded) => ({
		borderRadius: radius ? size * 0.5 : rounded ? 5 : undefined
	}),
	image: (size, radius, rounded) => ({
		width: size,
		height: size,
		backgroundColor: '#BCBEC1',
		alignItems: 'center',
		justifyContent: 'center',
		borderRadius: radius ? size * 0.5 : rounded ? 5 : undefined,
		overflow: 'hidden'
	}),
	textContainer: (size, bgColor, radius, rounded) => ({
		backgroundColor: bgColor,
		width: size,
		height: size,
		borderRadius: radius ? size * 0.5 : rounded ? 5 : undefined,
		alignItems: 'center',
		justifyContent: 'center'
	})
})

export default Component
