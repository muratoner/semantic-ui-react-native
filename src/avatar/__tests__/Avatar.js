import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Avatar from '../Avatar';

describe('Avatar Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Avatar />);

		let props = component.children().props();

		expect(component.find('Text')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);
		expect(props).toHaveProperty('children', undefined);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with textAutoColorMode is auto-light', () => {
		const component = shallow(
			<Avatar title="Murat ÖNER" textColorMode="auto-light" />
		);

		let props = component.children().props();

		expect(component.find('Text')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);
		expect(props).toHaveProperty('children', ['M', 'Ö']);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with title', () => {
		const component = shallow(<Avatar title="Murat ÖNER" />);

		let props = component.children().props();

		expect(component.find('Text')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);
		expect(props).toHaveProperty('children', ['M', 'Ö']);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with shortestTitle is false', () => {
		const component = shallow(
			<Avatar title="Murat ÖNER" shortestTitle={false} />
		);

		let props = component.children().props();

		expect(component.find('Text')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);
		expect(props).toHaveProperty('children', 'Murat ÖNER');

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with shortestTitle is false', () => {
		const component = shallow(
			<Avatar title="Murat ÖNER" shortestTitle={false} />
		);

		let props = component.children().props();

		expect(component.find('Text')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);
		expect(props).toHaveProperty('children', 'Murat ÖNER');

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with titleShowLimit is 2', () => {
		const component = shallow(
			<Avatar title="Murat Sancar Öner" titleShowLimit={2} />
		);

		let props = component.children().props();

		expect(component.find('Text')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);
		expect(props).toHaveProperty('children', ['M', 'S']);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with avatar image', () => {
		const component = shallow(
			<Avatar
				title="Murat Sancar Öner"
				source={{
					uri: 'https://i.pravatar.cc/50'
				}}
				titleShowLimit={2}
			/>
		);

		let imageBackground = component.find('ImageBackground').shallow();
		let image = imageBackground.find('Image').shallow();
		let text = imageBackground.find('Text').shallow();

		image.simulate('loadEnd');

		var imageStyles = image.get(0).props;

		expect(imageBackground).toHaveLength(1);
		expect(image).toHaveLength(1);
		expect(text).toHaveLength(1);
		expect(image.get(0).props).toHaveProperty('resizeMode', 'cover');
		expect(text.get(0).props).toHaveProperty('children', ['M', 'S']);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with avatar image', () => {
		const component = shallow(
			<Avatar
				title="Murat Sancar Öner"
				source={{
					uri: 'https://i.pravatar.cc/50'
				}}
				titleShowLimit={2}
			/>
		);

		let imageBackground = component.find('ImageBackground').shallow();
		let image = imageBackground.find('Image').shallow();
		let text = imageBackground.find('Text').shallow();

		image.simulate('loadEnd');

		var imageStyles = image.get(0).props;

		expect(imageBackground).toHaveLength(1);
		expect(image).toHaveLength(1);
		expect(text).toHaveLength(1);
		expect(image.get(0).props).toHaveProperty('resizeMode', 'cover');
		expect(text.get(0).props).toHaveProperty('children', ['M', 'S']);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with icon', () => {
		const component = shallow(<Avatar icon="user" />);

		let props = component.children().props();

		expect(component.find('Icon')).toHaveLength(1);
		expect(props).toHaveProperty('numberOfLines', 1);
		expect(props).toHaveProperty('adjustsFontSizeToFit', true);
		expect(props).toHaveProperty('minimumFontScale', 0.01);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with randomColor is false', () => {
		const component = shallow(
			<Avatar title={'Murat ÖNER'} withRandomColor={false} />
		);

		const text = component.find('Text');

		expect(text).toHaveLength(1);
		expect(text.props()).toHaveProperty('children', ['M', 'Ö']);
		expect(
			component.get(0).props.style.reduce((acc, cur) => Object.assign(acc, cur))
		).toHaveProperty('backgroundColor', '#b1b1b1');

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with circular is false', () => {
		const component = shallow(
			<Avatar
				source={{
					uri: 'https://i.pravatar.cc/50'
				}}
				circular={false}
			/>
		);

		const imageBackground = component.find('ImageBackground');

		expect(imageBackground).toHaveLength(1);
		expect(imageBackground.get(0).props.style).toHaveProperty(
			'borderRadius',
			undefined
		);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with onPress in Icon avatar', () => {
		const onPress = jest.fn();
		const component = shallow(<Avatar icon="user" onPress={onPress} />);

		component.simulate('press');

		expect(component).toHaveLength(1);
		expect(onPress).toBeCalled();
		expect(onPress).toBeCalledTimes(1);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with onPress in title avatar', () => {
		const onPress = jest.fn();
		const component = shallow(<Avatar title="Murat ÖNER" onPress={onPress} />);

		component.simulate('press');

		expect(component).toHaveLength(1);
		expect(onPress).toBeCalled();
		expect(onPress).toBeCalledTimes(1);

		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with onPress in image avatar', () => {
		const onPress = jest.fn();
		const component = shallow(
			<Avatar
				source={{
					uri: 'https://i.pravatar.cc/50'
				}}
				onPress={onPress}
			/>
		);

		component.simulate('press');

		expect(component).toHaveLength(1);
		expect(onPress).toBeCalled();
		expect(onPress).toBeCalledTimes(1);

		expect(toJson(component)).toMatchSnapshot();
	});
});
