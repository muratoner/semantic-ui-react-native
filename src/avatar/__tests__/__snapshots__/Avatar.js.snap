// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component should render with avatar image 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "borderRadius": 64,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": 64,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  />
</View>
`;

exports[`Avatar Component should render with avatar image 2`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "borderRadius": 64,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": 64,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  />
</View>
`;

exports[`Avatar Component should render with bordered prop 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#BCBEC1",
        "borderWidth": 1,
      },
      Object {
        "borderRadius": 64,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": 64,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.5)",
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </ImageBackground>
</View>
`;

exports[`Avatar Component should render with bordered prop 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {
        "borderColor": "#BCBEC1",
        "borderWidth": 1,
      },
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with circular is false 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "borderRadius": undefined,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": undefined,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.5)",
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </ImageBackground>
</View>
`;

exports[`Avatar Component should render with icon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#b1b1b1",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Icon
    0="u"
    1="s"
    2="e"
    3="r"
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "#fff",
        "fontSize": 18.285714285714285,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  />
</View>
`;

exports[`Avatar Component should render with onPress in Icon avatar 1`] = `
<ForwardRef
  onPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#b1b1b1",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Icon
    0="u"
    1="s"
    2="e"
    3="r"
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "#fff",
        "fontSize": 18.285714285714285,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  />
</ForwardRef>
`;

exports[`Avatar Component should render with onPress in image avatar 1`] = `
<ForwardRef
  onPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Array [
      Object {},
      Object {
        "borderRadius": 64,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": 64,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.5)",
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </ImageBackground>
</ForwardRef>
`;

exports[`Avatar Component should render with onPress in title avatar 1`] = `
<ForwardRef
  onPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</ForwardRef>
`;

exports[`Avatar Component should render with raised prop 1`] = `
<View
  style={
    Array [
      Object {
        "shadowColor": "rgba(0,0,0,.5)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
      },
      Object {
        "borderRadius": 64,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": 64,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.5)",
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </ImageBackground>
</View>
`;

exports[`Avatar Component should render with raised prop 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {
        "shadowColor": "rgba(0,0,0,.5)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
      },
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with randomColor is false 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#b1b1b1",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "#fff",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with rounded prop 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "borderRadius": 5,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.pravatar.cc/50",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BCBEC1",
        "borderRadius": 5,
        "height": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 32,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.5)",
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </ImageBackground>
</View>
`;

exports[`Avatar Component should render with rounded prop 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 5,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with shortestTitle is false 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Murat ÖNER
  </Text>
</View>
`;

exports[`Avatar Component should render with shortestTitle is false 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Murat ÖNER
  </Text>
</View>
`;

exports[`Avatar Component should render with textAutoColorMode is auto-light 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with textColorMode is light 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with title 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8bdf7",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0,0,0,.6)",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    Ö
  </Text>
</View>
`;

exports[`Avatar Component should render with titleShowLimit is 2 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4e2b75",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "#fff",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    M
    S
  </Text>
</View>
`;

exports[`Avatar Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {},
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    minimumFontScale={0.01}
    numberOfLines={1}
    style={
      Object {
        "color": "#fff",
        "fontSize": 200,
        "paddingHorizontal": 4.571428571428571,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  />
</View>
`;
